android-core/src/main/kotlin/com/mparticle/internal/messages/BaseMPMessage.kt

```diff
@@ -0,0 +1,74 @@
+package com.mparticle.internal.messages
+
+import android.location.Location
+import com.mparticle.internal.Constants
+import com.mparticle.internal.InternalSession
+import com.mparticle.internal.MPUtility
+import org.json.JSONException
+import org.json.JSONObject
+
+open class BaseMPMessage : JSONObject {
+    var mpId: Long = 0
+        private set
+
+    protected constructor()
+
+    constructor(builder: BaseMPMessageBuilder, session: InternalSession, location: Location?, mpId: Long) : super(builder, builder.keys) {
+        this.mpId = mpId
+        if (!has(Constants.MessageKey.TIMESTAMP)) {
+            put(Constants.MessageKey.TIMESTAMP, session.mLastEventTime)
+        }
+        if (Constants.MessageType.SESSION_START == builder.messageType) {
+            put(Constants.MessageKey.ID, session.mSessionID)
+        } else {
+            put(Constants.MessageKey.SESSION_ID, session.mSessionID)
+
+            if (session.mSessionStartTime > 0) {
+                put(Constants.MessageKey.SESSION_START_TIMESTAMP, session.mSessionStartTime)
+            }
+        }
+
+        if (!(Constants.MessageType.ERROR == builder.messageType &&
+                    Constants.MessageType.OPT_OUT != builder.messageType)
+        ) {
+            if (location != null) {
+                val locJSON = JSONObject()
+                locJSON.put(Constants.MessageKey.LATITUDE, location.latitude)
+                locJSON.put(Constants.MessageKey.LONGITUDE, location.longitude)
+                locJSON.put(Constants.MessageKey.ACCURACY, location.accuracy.toDouble())
+                put(Constants.MessageKey.LOCATION, locJSON)
+            }
+        }
+    }
+
+    val attributes: JSONObject?
+        get() = optJSONObject(Constants.MessageKey.ATTRIBUTES)
+
+    val timestamp: Long
+        get() {
+            try {
+                return getLong(Constants.MessageKey.TIMESTAMP)
+            } catch (_: JSONException) {
+            }
+            return 0
+        }
+
+    val sessionId: String
+        get() = if (Constants.MessageType.SESSION_START == messageType) {
```

Comment:
From a style perspective, should we standardize on the format used in line 48 (i.e. no `=` and using `{}` for any getting longer than one line? I know this technically works because the if/else is treated as a single expression, but it seems like it's cleaner/easier to read if we use {} around any multiline getters.

Reply:
I'm ok with using the `get() = ` format for single line getters